Makefile

```diff
@@ -8,29 +8,44 @@ OPENAPI_PATH=./vendor/k8s.io/kube-openapi
 
 VERSION?=latest
 
-.PHONY: all build-test-adapter test verify-gofmt gofmt verify test-adapter-container
-
+.PHONY: all
 all: build-test-adapter
-build-test-adapter: vendor test-adapter/generated/openapi/zz_generated.openapi.go
-	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) go build -o $(OUT_DIR)/$(ARCH)/test-adapter github.com/kubernetes-sigs/custom-metrics-apiserver/test-adapter
 
-vendor: tidy
-	go mod vendor
+.PHONY: build-test-adapter
+build-test-adapter: test-adapter/generated/openapi/zz_generated.openapi.go
+	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) go build -o $(OUT_DIR)/$(ARCH)/test-adapter github.com/kubernetes-sigs/custom-metrics-apiserver/test-adapter
 
-tidy:
-	go mod tidy
+test-adapter/generated/openapi/zz_generated.openapi.go: go.mod go.sum
+	go run $(OPENAPI_PATH)/cmd/openapi-gen/openapi-gen.go --logtostderr \
+	    -i k8s.io/metrics/pkg/apis/custom_metrics,k8s.io/metrics/pkg/apis/custom_metrics/v1beta1,k8s.io/metrics/pkg/apis/custom_metrics/v1beta2,k8s.io/metrics/pkg/apis/external_metrics,k8s.io/metrics/pkg/apis/external_metrics/v1beta1,k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/apimachinery/pkg/api/resource,k8s.io/apimachinery/pkg/version,k8s.io/api/core/v1 \
+	    -h ./hack/boilerplate.go.txt \
+	    -p ./test-adapter/generated/openapi \
+	    -O zz_generated.openapi \
+	    -o ./ \
+	    -r /dev/null
 
-test:
-	CGO_ENABLED=0 go test ./pkg/...
+.PHONY: gofmt
+gofmt:
+	./hack/gofmt-all.sh
 
+.PHONY: verify-gofmt
 verify-gofmt:
 	./hack/gofmt-all.sh -v
 
-gofmt:
-	./hack/gofmt-all.sh
+.PHONY: verify
+verify: verify-vendor verify-gofmt
+
+.PHONY: verify-vendor
+verify-vendor: go.mod
```

Comment:
Any reason to keep `vendor` directory? Is adding this an in between step needed before removal?

I'm not golang expert so maybe I'm missing something, but if vendor can be derived from `go.mod` and `go.sum` what is the benefit of keeping `vendor` directory in code?